InteractiveHtmlBom/ecad/genericjson.py

```diff
@@ -65,6 +67,48 @@ def _verify(self, pcb):
 
         return True
 
+    def _texts(self):
+        for layer in self.pcbdata['drawings'].values():
+            for side in layer.values():
+                for dwg in side:
+                    if 'text' in dwg:
+                        yield dwg
+
+    @staticmethod
+    def _remove_control_codes(s):
+        import unicodedata
+        return ''.join(c for c in s if unicodedata.category(c)[0] != "C")
+
+    def _parse_font_data(self):
+        font_parser = FontParser()
+        for dwg in self._texts():
+            if 'svgpath' not in dwg:
+                dwg['text'] = self._remove_control_codes(dwg['text'])
+                font_parser.parse_font_for_string(dwg['text'])
+
+        if font_parser.get_parsed_font().items():
+            self.pcbdata['font_data'] = {}
+            for (c, wl) in font_parser.get_parsed_font().items():
+                if c not in self.pcbdata['font_data']:
+                    self.pcbdata['font_data'][c] = wl
+
+    def _check_font_data(self):
+        self.mc = []
+        for dwg in self._texts():
+            dwg['text'] = self._remove_control_codes(dwg['text'])
+            self.mc += [c for c in dwg['text'] if 'svgpath' not in dwg and
+                        c not in self.pcbdata['font_data']]
+
+        self.mc = list(set(self.mc))  # get unique characters
+        s = ''.join(self.mc)
```

Comment:
This is hard to follow. Instead of accumulating a large array declare a set, keep adding chars to it (it will ignore elements it already contains) and in the end check if it's empty.